keras/backend/mxnet_backend.py

```diff
@@ -1614,11 +1614,14 @@ def equal(x, y):
     if isinstance(y, KerasSymbol):
         y = y.symbol
         scalar = True
-    if scalar:
+    if scalar and x.infer_shape() == y.infer_shape():
```

Comment:
There is no need for this if..else. broadcast_equal will work for both same shape and shapes that require broadcasting.

Reply:
Same for all other operators below.